mm2src/mm2_main/src/lp_swap/maker_swap.rs

```diff
@@ -1926,19 +1926,33 @@ impl MakerSavedSwap {
         }
     }
 
+    // TODO: Adjust for private coins when/if they are braodcasted
+    // TODO: Adjust for HD wallet when completed
     pub fn swap_pubkeys(&self) -> Result<SwapPubkeys, String> {
-        match self.events.first() {
+        let mut pubkeys = SwapPubkeys {
+            maker: None,
+            taker: None,
+        };
+
+        match &self.events.first() {
             Some(event) => match &event.event {
-                // TODO: Adjust for private coins when/if they are braodcasted
-                // TODO: Adjust for HD wallet when completed
-                MakerSwapEvent::Started(data) => Ok(SwapPubkeys::new(
-                    data.my_persistent_pub.to_string(),
-                    data.taker.to_string(),
-                )),
-                _ => ERR!("First swap event must be Started"),
+                MakerSwapEvent::Started(started) => pubkeys.maker = Some(started.my_persistent_pub.to_string()),
+                _ => return ERR!("First swap event must be Started"),
             },
-            None => ERR!("Can't get maker/taker pubkey, events are empty"),
-        }
+            None => return ERR!("Can't get maker's pubkey when events are empty"),
+        };
+
+        match self.events.get(1) {
+            Some(event) => match &event.event {
+                MakerSwapEvent::Negotiated(neg) => {
+                    pubkeys.taker = neg.taker_coin_htlc_pubkey.map(|key| key.to_string());
```

Comment:
You can return an error if `taker_coin_htlc_pubkey` is `None` since it should be always `Some`
https://github.com/KomodoPlatform/atomicDEX-API/blob/26e33bc2c1bef2603abd2465aca9a806cfd5c356/mm2src/mm2_main/src/lp_swap/maker_swap.rs#L678